pandas/tests/indexing/test_indexing.py

```diff
@@ -717,6 +717,13 @@ def test_index_type_coercion(self):
                     idxr(s2)['0'] = 0
                     assert s2.index.is_object()
 
+    def test_empty_index_for_empty_dataframe(self):
+        index = pd.Index([], name='idx')
+        result = pd.DataFrame(columns=['A'], index=index)
+        result['A'] = []
+        expected = pd.DataFrame(columns=['A'], index=index)
+        tm.assert_index_equal(result.index, expected.index)
```

Comment:
use tm.assert_frame_equal(result, expected)

Reply:
@jreback frame are not equal, only their index is, snippet
```
index = pd.Index([], name='idx')
result = pd.DataFrame(columns=['A'], index=index)
result['A'] = []
expected = pd.DataFrame(columns=['A'], index=index)
tm.assert_frame_equal(result, expected)
```

results in ```Attribute "dtype" are different
[left]:  float64
[right]: object```

Reply:
construct the expected so that it has the same dtypes (e.g. pass dtype=float64)